### ISSUE TITLE
panic: duplicate metrics collector registration attempted

### ISSUE THREAD

**Issue by compilenix** [grafana/loki#6797]
**Describe the bug**
When `compactor.deletion_mode` is set to `filter-and-delete`, using the [default docker-compose getting started guide](https://grafana.com/docs/loki/latest/installation/docker/#install-with-docker-compose). loki does not start.

My aim is to configure loki to delete all log data older then 48 hours.

**To Reproduce**
1. Set `compactor.deletion_mode` to `filter-and-delete` in `/etc/loki/local-config.yaml`
2. Start Loki (version: 2.6.0 image id: f6f2aab1b4ab)

**Expected behavior**
Loki to start or to show what is wrong with the config.

**Environment:**
 - Infrastructure: ubuntu 20.04 vmware vm
 - Deployment tool: docker-compose

**Screenshots, Promtail config, or terminal output**
`/etc/loki/local-config.yaml`
```
auth_enabled: false

server:
  http_listen_port: 3100
  register_instrumentation: false

limits_config:
  ingestion_rate_mb: 100
  max_streams_per_user: 100000
  retention_period: 48h

compactor:
  retention_enabled: true
  deletion_mode: filter-and-delete

common:
  path_prefix: /loki
  storage:
    filesystem:
      chunks_directory: /loki/chunks
      rules_directory: /loki/rules
  replication_factor: 1
  ring:
    kvstore:
      store: inmemory

schema_config:
  configs:
    - from: 2020-10-24
      store: boltdb-shipper
      object_store: filesystem
      schema: v11
      index:
        prefix: index_
        period: 24h

ruler:
  alertmanager_url: http://alertmanager:9093


# By default, Loki will send anonymous, but uniquely-identifiable usage and configuration
# analytics to Grafana Labs. These statistics are sent to https://stats.grafana.org/
#
# Statistics help us better understand how Loki is used, and they show us performance
# levels for most users. This helps us prioritize features and documentation.
# For more information on what's sent, look at
# https://github.com/grafana/loki/blob/main/pkg/usagestats/stats.go
# Refer to the buildReport method to see what goes into a report.
#
# If you would like to disable reporting, uncomment the following lines:
analytics:
  reporting_enabled: false
```

`docker-compose.yml`
```yml
version: '2.4'

volumes:
  loki_data: {}

services:
  loki:
    image: grafana/loki:2.6.0
    ports:
      - 3100:3100
    command: -config.file=/etc/loki/local-config.yaml
    volumes:
      - loki_data:/loki
      - ./config.yml:/etc/loki/local-config.yaml
```

`docker-compose logs loki`
```
loki_1    | level=info ts=2022-07-28T14:08:42.696937463Z caller=main.go:103 msg="Starting Loki" version="(version=2.6.0, branch=HEAD, revision=b92f113cb)"
loki_1    | level=info ts=2022-07-28T14:08:42.697296142Z caller=server.go:260 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
loki_1    | level=warn ts=2022-07-28T14:08:42.702832773Z caller=experimental.go:20 msg="experimental feature in use" feature="In-memory (FIFO) cache - chunksfifocache"
loki_1    | level=info ts=2022-07-28T14:08:42.704126122Z caller=table_manager.go:381 msg="loading local table index_19201"
loki_1    | ts=2022-07-28T14:08:42.704528477Z caller=spanlogger.go:80 level=info msg="building index list cache"
loki_1    | ts=2022-07-28T14:08:42.709005802Z caller=spanlogger.go:80 level=info msg="index list cache built" duration=4.457083ms
loki_1    | level=info ts=2022-07-28T14:08:42.711676096Z caller=table_manager.go:134 msg="uploading tables"
loki_1    | level=info ts=2022-07-28T14:08:42.718603127Z caller=table_manager.go:252 msg="query readiness setup completed" duration=1.75µs distinct_users_len=0
loki_1    | level=info ts=2022-07-28T14:08:42.718631609Z caller=shipper.go:124 msg="starting index shipper in RW mode"
loki_1    | level=info ts=2022-07-28T14:08:42.719040925Z caller=table_manager.go:226 msg="loading table index_19201"
loki_1    | level=info ts=2022-07-28T14:08:42.720887278Z caller=table.go:319 msg="handing over indexes to shipper index_19201"
loki_1    | level=info ts=2022-07-28T14:08:42.720924526Z caller=table.go:335 msg="finished handing over table index_19201"
loki_1    | level=info ts=2022-07-28T14:08:42.720959023Z caller=shipper_index_client.go:79 msg="starting boltdb shipper in RW mode"
loki_1    | level=info ts=2022-07-28T14:08:42.724766452Z caller=table_manager.go:167 msg="handing over indexes to shipper"
loki_1    | level=info ts=2022-07-28T14:08:42.724832248Z caller=table.go:319 msg="handing over indexes to shipper index_19201"
loki_1    | level=info ts=2022-07-28T14:08:42.724839875Z caller=table.go:335 msg="finished handing over table index_19201"
loki_1    | level=info ts=2022-07-28T14:08:42.7262032Z caller=mapper.go:46 msg="cleaning up mapped rules directory" path=/loki/rules-temp
loki_1    | level=error ts=2022-07-28T14:08:42.726239889Z caller=mapper.go:50 msg="unable to read rules directory" path=/loki/rules-temp err="open /loki/rules-temp: no such file or directory"
loki_1    | panic: duplicate metrics collector registration attempted
loki_1    | 
loki_1    | goroutine 1 [running]:
loki_1    | github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0x21d3ff7, {0xc0000b7720, 0x1, 0x0})
loki_1    | 	/src/loki/vendor/github.com/prometheus/client_golang/prometheus/registry.go:403 +0x7f
loki_1    | github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounter({{0x26455d0, 0xc00012ca00}}, {{0x21d3ff7, 0x4}, {0x0, 0x0}, {0x220abb2, 0x1c}, {0x22541ce, 0x35}, ...})
loki_1    | 	/src/loki/vendor/github.com/prometheus/client_golang/prometheus/promauto/auto.go:265 +0x102
loki_1    | github.com/grafana/loki/pkg/storage/stores/shipper/compactor/deletion.newDeleteRequestClientMetrics({0x26455d0, 0xc00012ca00})
loki_1    | 	/src/loki/pkg/storage/stores/shipper/compactor/deletion/metrics.go:16 +0x9c
loki_1    | github.com/grafana/loki/pkg/storage/stores/shipper/compactor/deletion.NewDeleteRequestsClient({0xc00028dea8, 0x400}, {0x2629880, 0xc000662150}, {0x26455d0, 0xc00012ca00}, {0x0, 0x0, 0x203000})
loki_1    | 	/src/loki/pkg/storage/stores/shipper/compactor/deletion/delete_requests_client.go:66 +0xce
loki_1    | github.com/grafana/loki/pkg/loki.(*Loki).deleteRequestsClient(0xc0005d2000)
loki_1    | 	/src/loki/pkg/loki/modules.go:992 +0x1ce
loki_1    | github.com/grafana/loki/pkg/loki.(*Loki).initQuerier(0xc0005d2000)
loki_1    | 	/src/loki/pkg/loki/modules.go:244 +0x57
loki_1    | github.com/grafana/dskit/modules.(*Manager).initModule(0xc000277c98, {0x21d3208, 0xc00089c530}, 0x807ee7, 0x0)
loki_1    | 	/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:120 +0x224
loki_1    | github.com/grafana/dskit/modules.(*Manager).InitModuleServices(0xc00089c580, {0xc0000b6e60, 0x1, 0x20844c0})
loki_1    | 	/src/loki/vendor/github.com/grafana/dskit/modules/modules.go:92 +0x10c
loki_1    | github.com/grafana/loki/pkg/loki.(*Loki).Run(0xc0005d2000, {0xc000561d80})
loki_1    | 	/src/loki/pkg/loki/loki.go:337 +0x56
loki_1    | main.main()
loki_1    | 	/src/loki/cmd/loki/main.go:105 +0xdc5
prometheus_loki_1 exited with code 2
```

If you need anything else, just let me know :)

**Comment by MichelHollands** [comment]
Hello, this is a known issue which we have fixed in version 2.6.1.

Also note that to enable retention you don't need change the deletion_mode. Setting `deletion_mode` to anything other than `disabled` will enable the [Deletion API](https://grafana.com/docs/loki/latest/operations/storage/logs-deletion/). This is not required for [retention](https://grafana.com/docs/loki/latest/operations/storage/retention/) on it's own.

**Comment by compilenix** [comment]
Thank you very much!
